apps/omg/lib/omg/output/fungible_morevp_token.ex

```diff
@@ -14,26 +14,29 @@
 
 defmodule OMG.Output.FungibleMoreVPToken do
   @moduledoc """
-  Representation of the payment transaction output of a fungible token `currency`
+  Representation of the payment transaction output of a fungible token `currency`.
+  Fungible token outputs could have different `type_marker` but as long as they share the same behaviour
+  they are indistinguishable from code perspective.
   """
   alias OMG.Crypto
-  defstruct [:owner, :currency, :amount]
+  defstruct [:owner, :currency, :amount, :type_marker]
 
   @type t :: %__MODULE__{
           owner: Crypto.address_t(),
           currency: Crypto.address_t(),
-          amount: non_neg_integer()
+          amount: non_neg_integer(),
+          type_marker: binary()
```

Comment:
Is `type_marker` a `binary` or integer? Asking for confirmation as I've also been encoding this as a binary, but it seems like integer is also valid.

Reply:
good catch - approach to this was lax due to RLP being lax in the aspect of decoding `<<1>>`. Now as contracts are more strict on this, we should probably conform stricter and handle type (markers) as integers on the Elixir end as well.

But this is more of a "least astonishment" kind of conformance, since these markers do not have any inherent "meaning" attached to their value.

Reply:
As @InoMurko pointed out RLP-encoding both integer `1` and binary `<<1>>` we get the same `<<1>>` result. Solidity decodes it to `uint256` and hence erlang support for bignums I think I switch to non_neg_integer here.